### ISSUE TITLE
Latest metasploit-framework not working inside docker (broken pipe)

### ISSUE THREAD

**Issue by FireFart** [rapid7/metasploit-framework#8211]
Currently getting the following error message when building and running the docker image. Tested on OSX and Fedora:

```
/usr/src/metasploit-framework/lib/msf/core/modules/external/bridge.rb:84:in `write': Broken pipe (Errno::EPIPE)
	from /usr/src/metasploit-framework/lib/msf/core/modules/external/bridge.rb:84:in `write_message'
	from /usr/src/metasploit-framework/lib/msf/core/modules/external/bridge.rb:67:in `send'
	from /usr/src/metasploit-framework/lib/msf/core/modules/external/bridge.rb:55:in `send_receive'
	from /usr/src/metasploit-framework/lib/msf/core/modules/external/bridge.rb:47:in `describe'
	from /usr/src/metasploit-framework/lib/msf/core/modules/external/bridge.rb:11:in `meta'
	from /usr/src/metasploit-framework/lib/msf/core/modules/external/shim.rb:8:in `generate'
	from /usr/src/metasploit-framework/lib/msf/core/modules/loader/executable.rb:85:in `read_module_content'
	from /usr/src/metasploit-framework/lib/msf/core/modules/loader/base.rb:126:in `load_module'
	from /usr/src/metasploit-framework/lib/msf/core/modules/loader/base.rb:237:in `block in load_modules'
	from /usr/src/metasploit-framework/lib/msf/core/modules/loader/executable.rb:51:in `block (2 levels) in each_module_reference_name'
	from /usr/local/bundle/gems/rex-core-0.1.9/lib/rex/file.rb:133:in `block in find'
	from /usr/local/bundle/gems/rex-core-0.1.9/lib/rex/file.rb:132:in `catch'
	from /usr/local/bundle/gems/rex-core-0.1.9/lib/rex/file.rb:132:in `find'
	from /usr/src/metasploit-framework/lib/msf/core/modules/loader/executable.rb:42:in `block in each_module_reference_name'
	from /usr/src/metasploit-framework/lib/msf/core/modules/loader/executable.rb:31:in `foreach'
	from /usr/src/metasploit-framework/lib/msf/core/modules/loader/executable.rb:31:in `each_module_reference_name'
	from /usr/src/metasploit-framework/lib/msf/core/modules/loader/base.rb:236:in `load_modules'
	from /usr/src/metasploit-framework/lib/msf/core/module_manager/loading.rb:119:in `block in load_modules'
	from /usr/src/metasploit-framework/lib/msf/core/module_manager/loading.rb:117:in `each'
	from /usr/src/metasploit-framework/lib/msf/core/module_manager/loading.rb:117:in `load_modules'
	from /usr/src/metasploit-framework/lib/msf/core/module_manager/module_paths.rb:41:in `block in add_module_path'
	from /usr/src/metasploit-framework/lib/msf/core/module_manager/module_paths.rb:40:in `each'
	from /usr/src/metasploit-framework/lib/msf/core/module_manager/module_paths.rb:40:in `add_module_path'
	from /usr/src/metasploit-framework/lib/msf/base/simple/framework/module_paths.rb:50:in `block in init_module_paths'
	from /usr/src/metasploit-framework/lib/msf/base/simple/framework/module_paths.rb:49:in `each'
	from /usr/src/metasploit-framework/lib/msf/base/simple/framework/module_paths.rb:49:in `init_module_paths'
	from /usr/src/metasploit-framework/lib/msf/ui/console/driver.rb:219:in `initialize'
	from /usr/src/metasploit-framework/lib/metasploit/framework/command/console.rb:62:in `new'
	from /usr/src/metasploit-framework/lib/metasploit/framework/command/console.rb:62:in `driver'
	from /usr/src/metasploit-framework/lib/metasploit/framework/command/console.rb:48:in `start'
	from /usr/src/metasploit-framework/lib/metasploit/framework/command/base.rb:82:in `start'
	from ./msfconsole:48:in `<main>'
```

**Comment by FireFart** [comment]
Pinging @acammack-r7 because he wrote the conflicting code. Would be great if you can have a look.

**Comment by busterb** [comment]
It's kind of weird how the error only occurs the 2nd time you run the docker image, not the first, so I didn't notice it at first. Also including a bogus hashbang path in the module didn't reproduce the issue either. Should be easy to fix though.

**Comment by busterb** [comment]
Hmm, if the module fails to load quickly enough, the exception does get caught quickly and logged to framework.log:

```
[04/09/2017 11:34:42] [e(0)] core: metasploit-framework/modules/exploits/linux/smtp/haraka.rb failed to load due to the following error:
Errno::ENOENT
No such file or directory @ rb_sysopen - /Users/bcook/rapid7/metasploit-framework/modules/exploits/linux/smtp/haraka.rb
```

**Comment by FireFart** [comment]
@busterb oh I missed this log entry. It looks like there is a python module with this name checked in, but the framework assumes the .rb extension:
https://github.com/rapid7/metasploit-framework/blob/master/modules/exploits/linux/smtp/haraka.py

**Comment by FireFart** [comment]
@busterb If I delete the file the container starts as it should

**Comment by busterb** [comment]
I just submitted a PR to handle the exception when the module disconnects while running. I think we expected the ENOENT check to work inside of docker too, but it didn't for some reason, maybe due to different behavior of ruby or /usr/bin/env

**Comment by busterb** [comment]
The root cause appears to be the difference in how /usr/bin/env works in busybox vs native OS X / coreutils. We probably wouldn't have noticed if the docker image were based on something other than alpine

lrwxrwxrwx    1 root     root            12 Mar 30 22:29 /usr/bin/env -> /bin/busybox

**Comment by FireFart** [comment]
@busterb seeing the issues again in the latest docker image when running exploits:
```
[01/23/2018 16:08:06] [e(0)] core: Unexpected output running /usr/src/metasploit-framework/modules/auxiliary/dos/http/slowloris.py:
env: can't execute 'python': No such file or directory

[01/23/2018 16:08:06] [e(0)] core: Unable to load module /usr/src/metasploit-framework/modules/auxiliary/dos/http/slowloris.py NoMethodError undefined method `params' for nil:NilClass
[01/23/2018 16:08:07] [e(0)] core: /usr/src/metasploit-framework/modules/exploits/linux/smtp/haraka.rb failed to load due to the following error:
Errno::ENOENT
No such file or directory @ rb_sysopen - /usr/src/metasploit-framework/modules/exploits/linux/smtp/haraka.rb
Call stack:
/usr/src/metasploit-framework/lib/msf/core/modules/loader/directory.rb:78:in `initialize'
/usr/src/metasploit-framework/lib/msf/core/modules/loader/directory.rb:78:in `open'
/usr/src/metasploit-framework/lib/msf/core/modules/loader/directory.rb:78:in `read_module_content'
/usr/src/metasploit-framework/lib/msf/core/modules/loader/base.rb:126:in `load_module'
/usr/src/metasploit-framework/lib/msf/core/module_manager/cache.rb:91:in `block in load_cached_module'
/usr/src/metasploit-framework/lib/msf/core/module_manager/cache.rb:86:in `each'
/usr/src/metasploit-framework/lib/msf/core/module_manager/cache.rb:86:in `load_cached_module'
/usr/src/metasploit-framework/lib/msf/core/module_set.rb:45:in `create'
/usr/src/metasploit-framework/lib/msf/core/db_manager/module_cache.rb:318:in `block (3 levels) in update_all_module_details'
/usr/src/metasploit-framework/lib/msf/core/db_manager/module_cache.rb:316:in `each'
/usr/src/metasploit-framework/lib/msf/core/db_manager/module_cache.rb:316:in `block (2 levels) in update_all_module_details'
/usr/src/metasploit-framework/lib/msf/core/db_manager/module_cache.rb:313:in `each'
/usr/src/metasploit-framework/lib/msf/core/db_manager/module_cache.rb:313:in `block in update_all_module_details'
/usr/local/bundle/gems/activerecord-4.2.10/lib/active_record/connection_adapters/abstract/connection_pool.rb:292:in `with_connection'
/usr/src/metasploit-framework/lib/msf/core/db_manager/module_cache.rb:276:in `update_all_module_details'
/usr/src/metasploit-framework/lib/msf/ui/console/command_dispatcher/db.rb:1709:in `block in cmd_db_connect'
/usr/src/metasploit-framework/lib/msf/core/thread_manager.rb:100:in `block in spawn'
[01/23/2018 16:08:07] [e(0)] core: Unexpected output running /usr/src/metasploit-framework/modules/exploits/linux/smtp/haraka.py:
env: can't execute 'python2.7': No such file or directory

[01/23/2018 16:08:07] [e(0)] core: Unable to load module /usr/src/metasploit-framework/modules/exploits/linux/smtp/haraka.py NoMethodError undefined method `params' for nil:NilClass
```

**Comment by busterb** [comment]
Bumped docker dependencies to fix this in #9470
